src/internet_identity/src/storage.rs

```diff
@@ -635,3 +630,24 @@ impl fmt::Display for StorageError {
         }
     }
 }
+
+/// Helper module to hide internal memory of the memory size reference.
+mod size_ref {
+    use ic_stable_structures::Memory;
+
+    /// Struct that holds a memory with the sole purpose to provide a function to get
+    /// the size of the memory.
+    pub struct MemorySizeRef<M: Memory> {
```

Comment:
nit: maybe `MemoryWrapper` or something similar?  `MemorySizeRef` sounds very cryptic, and may become "outdated" if another method is added to the impl 😉